neps/nep-0488.md

```diff
@@ -0,0 +1,1537 @@
+
+The constants used to compute $x_{den}$ are as follows:
+
+- $$k_{(2,0)} = 4002409555221667393417789825735904156556882819939007885332058136124031650490837864442687629129015664037894272559715 \cdot I$$
+
+- $$k_{(2,1)} = 12 + 4002409555221667393417789825735904156556882819939007885332058136124031650490837864442687629129015664037894272559775 \cdot I$$
+
+The constants used to compute $y_{num}$ are as follows:
+
+- $$k_{(3,0)} = 3261222600550988246488569487636662646083386001431784202863158481286248011511053074731078808919938689216061999863558 +\\
+             3261222600550988246488569487636662646083386001431784202863158481286248011511053074731078808919938689216061999863558 \cdot I$$
+
+- $$k_{(3,1)} = 889424345604814976315064405719089812568196182208668418962679585805340366775741747653930584250892369786198727235518 \cdot I$$
+
+- $$k_{(3,2)} = 2668273036814444928945193217157269437704588546626005256888038757416021100327225242961791752752677109358596181706524 +
+             1334136518407222464472596608578634718852294273313002628444019378708010550163612621480895876376338554679298090853263 \cdot I$$
+
+- $$k_{(3,3)} = 2816510427748580758331037284777117739799287910327449993381818688383577828123182200904113516794492504322962636245776$$
+
+The constants used to compute $y_{den}$ are as follows:
+
+- $$k_{(4,0)} = 4002409555221667393417789825735904156556882819939007885332058136124031650490837864442687629129015664037894272559355 + 
+             4002409555221667393417789825735904156556882819939007885332058136124031650490837864442687629129015664037894272559355 \cdot I$$
+
+- $$k_{(4,1)} = 4002409555221667393417789825735904156556882819939007885332058136124031650490837864442687629129015664037894272559571 \cdot I$$
+
+- $$k_{(4,2)} = 18 + 4002409555221667393417789825735904156556882819939007885332058136124031650490837864442687629129015664037894272559769 \cdot I$$
+
+**Step 3:** Effective cofactor:
+
+- $$h_{eff} = 209869847837335686905080341498658477663839067235703451875306851526599783796572738804459333109033834234622528588876978987822447936461846631641690358257586228683615991308971558879306463436166481$$
+
+### Curve points encoding
+
+#### Bool as output
+
+The bool is encoded as a little-endian `u64` type in Rust. The `true` value is encoded as `1` and `false` is encoded as `0`. All other values of `u64` are not allowed and should be interpreted as incorrect. Encoding is the same as for BN254 implementation in nearcore[^50].
+
+#### Sign
+
+The sign of the point on the elliptic curve is encoded as `u8` type in Rust with two possible values: `0` and `1`. `0` for positive sign, and `1` for negative sign. All other value of `u8` is not allowed and should be interpreted as incorrect.
+
+#### Scalar
+
+The scalar value is encoded as a big-endian `[u8;32]`. All possible bytes combination is allowed. Encoding is similar with BN254 implementation in nearcore[^50], but `big-endian` encoding is used instead of `little-endian` as in EIP-2537[^15].
```

Comment:
I believe that this particular encoding was chosen to be consistent with existing standards. Otherwise, it would be necessary to re-encode any actual inputs (reverse the order of bytes, set/clear some bits, etc.), and the current approach seems to be more developer-friendly.

Reply:
I guess it’s a matter of choice between being consistent with existing standards, and being consistent with our own codebase. Considering how I regard PHP, I’d lean towards being consistent with our own codebase rather than being consistent with external habits, but I can understand opinions vary on this matter.

Reply:
I agree with @abacabadabacaba that in this case it might be better to be consistent with existing standards, especially given that one of the goals of this NEP is to improve interoperability.

Reply:
@Ekleog-NEAR Actually, I just double checked it and it really should be a little-endian. Standart which I whould like to follow says nothing about scalars (https://github.com/supranational/blst#serialization-format). In some libraries it is possible to decode both from le and be. And blst is using le for scalars, just checked it. 

I will fix this place. Thank you for notice.

Reply:
fixed: https://github.com/near/NEPs/pull/488/commits/7c5cb3b4c8ffea0d13f0a0ec6127923f288fc9dc